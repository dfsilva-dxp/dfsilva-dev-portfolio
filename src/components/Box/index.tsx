import React, { ReactNode, useLayoutEffect, useRef } from 'react';
import gsap from 'gsap';

import { Flex } from '@/components';

import * as S from './styles';
import { DefaultTheme } from '@/styles';

interface IBoxProps {
  firstChild: ReactNode;
  lastChild: ReactNode;
  icon?: ReactNode;
  alignWrapper?: 'center' | 'flex-end';
  justifyChildren?: 'center' | 'space-between';
  padding?: '3.2rem' | '3.2rem 6rem';
}

const Box = ({
  firstChild,
  lastChild,
  icon,
  alignWrapper = 'flex-end',
  justifyChildren = 'space-between',
  padding = '3.2rem 6rem'
}: IBoxProps) => {
  const splitElementRef = useRef<HTMLDivElement>(null);
  const gridBoxRef = useRef<HTMLDivElement>(null);
  const ctx = useRef<gsap.Context | null>();

  const onEnter = () => {
    if (ctx.current) {
      ctx.current.onEnter();
    }
  };

  const onLeave = () => {
    if (ctx.current) {
      ctx.current.onLeave();
    }
  };

  useLayoutEffect(() => {
    ctx.current = gsap.context((self) => {
      const tl = gsap
        .timeline({
          paused: true,
          defaults: {
            ease: 'power1.inOut',
            duration: 0.4
          }
        })
        .to(gridBoxRef.current, {
          background: DefaultTheme.colors.text.heading
        })
        .to(
          splitElementRef.current!.children,
          {
            yPercent: -100,
            stagger: -0.05
          },
          0
        );
      self.add('onEnter', () => {
        tl.play();
      });

      self.add('onLeave', () => {
        tl.reverse();
      });
    });

    return () => {
      ctx.current && ctx.current.revert();
    };
  }, []);

  return (
    <S.BoxWrapper
      align={alignWrapper}
      ref={gridBoxRef}
      onMouseEnter={onEnter}
      onMouseLeave={onLeave}
      style={{
        padding
      }}
    >
      <Flex
        align="flex-end"
        justify={justifyChildren}
        style={{
          width: '100%'
        }}
      >
        <S.BoxSplitWrapper ref={splitElementRef}>
          <S.BoxSplitChildren>{firstChild}</S.BoxSplitChildren>
          <S.BoxSplitChildren>{lastChild}</S.BoxSplitChildren>
        </S.BoxSplitWrapper>
        {!!icon && icon}
      </Flex>
    </S.BoxWrapper>
  );
};

export default Box;
