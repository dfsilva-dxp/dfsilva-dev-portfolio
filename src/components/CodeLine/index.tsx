import { ReactNode } from 'react';
import { Fira_Code } from 'next/font/google';

const firaCode = Fira_Code({
  weight: ['500'],
  subsets: ['latin'],
  display: 'swap'
});

import * as S from './styles';

interface ICodeLineProps {
  lineNumber: string;
  children: ReactNode;
}

const CodeLine = ({ children, lineNumber }: ICodeLineProps) => {
  return (
    <S.CodeLineContent className={firaCode.className}>
      <S.CodeLineNumber>{lineNumber}</S.CodeLineNumber>

      <S.CodeLineScript>{children}</S.CodeLineScript>
    </S.CodeLineContent>
  );
};

export default CodeLine;
