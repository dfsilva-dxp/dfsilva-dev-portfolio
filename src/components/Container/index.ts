'use client';

import styled, { css } from 'styled-components';

const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding-right: 1.6rem;
    padding-left: 1.6rem;
    margin-right: auto;
    margin-left: auto;

    @media (min-width: 576px) {
      max-width: ${theme.grid.sm};
    }
    @media (min-width: 768px) {
      max-width: ${theme.grid.md};
    }
    @media (min-width: 992px) {
      max-width: ${theme.grid.lg};
    }
    @media (min-width: 1200px) {
      max-width: ${theme.grid.xlg};
    }
    @media (min-width: 1400px) {
      max-width: ${theme.grid.xxlg};
    }
  `}
`;

export default Container;
