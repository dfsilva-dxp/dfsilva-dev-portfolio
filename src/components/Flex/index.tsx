import {
  ForwardRefRenderFunction,
  HTMLAttributes,
  ReactNode,
  forwardRef
} from 'react';

import * as S from './styles';

export type FlexType = {
  display?: 'none' | 'inline-flex' | 'flex';
  direction?: 'row' | 'column' | 'row-reverse' | 'column-reverse';
  align?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline';
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  content?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch';
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
  gap?:
    | '0'
    | '0.4rem'
    | '0.8rem'
    | '1rem'
    | '1.6rem'
    | '2rem'
    | '2.4rem'
    | '2.8rem'
    | '3rem'
    | '3.2rem'
    | '3.4rem'
    | '3.8rem'
    | '4.5rem'
    | '6rem';
};

type FlexDivType = HTMLAttributes<HTMLDivElement>;

type IFlexProps = {
  children: ReactNode;
} & FlexType &
  FlexDivType;

const Flex: ForwardRefRenderFunction<HTMLDivElement, IFlexProps> = (
  {
    children,
    display = 'flex',
    direction = 'row',
    align = 'flex-start',
    justify = 'flex-start',
    content = 'stretch',
    wrap = 'wrap',
    gap = '0',
    ...props
  },
  ref
) => {
  return (
    <S.FlexContent
      ref={ref}
      display={display}
      direction={direction}
      align={align}
      justify={justify}
      content={content}
      wrap={wrap}
      gap={gap}
      {...props}
    >
      {children}
    </S.FlexContent>
  );
};

export const FlexComponent = forwardRef(Flex);
