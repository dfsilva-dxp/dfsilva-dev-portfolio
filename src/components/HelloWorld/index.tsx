import * as S from './styles';

const HelloWorld = () => {
  return (
    <>
      <S.ConsoleElement>console</S.ConsoleElement>
      <S.DotElement>.</S.DotElement>
      <S.LogElement>log</S.LogElement>
      <S.ElementOpensParenthesis>{'('}</S.ElementOpensParenthesis>
      <S.StringElement>{'"hello world"'}</S.StringElement>
      <S.ElementClosesParenthesis>{')'}</S.ElementClosesParenthesis>
      <S.SemicolonElement>;</S.SemicolonElement>
    </>
  );
};

export default HelloWorld;
