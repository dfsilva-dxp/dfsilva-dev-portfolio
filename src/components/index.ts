export { default as Container } from './Container';
export { default as Flex } from './Flex';
export { default as Header } from './Header';
export { default as Hero } from './Hero';
export { default as Loader } from './Loader';
export { default as Tag } from './Tag';
