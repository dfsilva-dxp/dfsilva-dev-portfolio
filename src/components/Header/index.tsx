import { useState } from 'react';
import { AnimatePresence, motion } from 'framer-motion';

import { Container, Flex } from '@/components';

import * as S from './styles';
import {
  content,
  fullscreen,
  header,
  navButtonAnimation
} from './motion-config';

const Header = () => {
  const [isActive, setIsActive] = useState(false);

  const toggleMenu = () => {
    setTimeout(() => setIsActive((current) => !current), 300);
  };

  return (
    <>
      <motion.div
        variants={header}
        initial="initial"
        animate="animate"
        className="motion-container"
      >
        <S.HeaderContent>
          <Flex align="center" justify="flex-end">
            <motion.div variants={navButtonAnimation}>
              <S.ToggleButton type="button" onClick={toggleMenu}>
                {isActive ? 'Voltar' : 'Navegue'}
              </S.ToggleButton>
            </motion.div>
          </Flex>
        </S.HeaderContent>
      </motion.div>

      <AnimatePresence>
        {isActive && (
          <S.Overlay
            variants={fullscreen}
            initial="initial"
            animate="animate"
            exit="exit"
          >
            <Container>
              <motion.div
                variants={content}
                initial="initial"
                animate="open"
                exit="initial"
                className="motion-container"
              >
                <Flex align="center" justify="center">
                  <Flex direction="column" gap="1.6rem">
                    <strong>Menu</strong>
                  </Flex>
                </Flex>
              </motion.div>
            </Container>
          </S.Overlay>
        )}
      </AnimatePresence>
    </>
  );
};

export default Header;
