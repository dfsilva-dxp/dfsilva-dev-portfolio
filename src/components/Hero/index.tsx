import Image from 'next/image';
import { motion } from 'framer-motion';
import {
  GithubLogo,
  InstagramLogo,
  LinkedinLogo,
  TwitterLogo
} from 'phosphor-react';

import { Container, Flex, Tag } from '@/components';

import * as S from './styles';

import { hero, letterAnimation } from './motion-config';

const Hero = () => {
  return (
    <S.HeroContent>
      <Container>
        <motion.div
          variants={hero}
          initial="initial"
          animate="animate"
          className="motion-container"
        >
          <S.SocialMediaContent>
            <Flex align="center" gap="1rem" direction="column">
              <Flex align="center" gap="1rem" direction="column">
                <a
                  href="https://github.com/dfsilvadev"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <GithubLogo size={18} weight="bold" />
                </a>

                <a
                  href="https://www.instagram.com/dfsilva.dev/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <InstagramLogo size={18} weight="bold" />
                </a>

                <a
                  href="https://www.linkedin.com/in/dfsilva-dev"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <LinkedinLogo size={18} weight="bold" />
                </a>

                <a
                  href="https://twitter.com/dfsilvadev"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <TwitterLogo size={18} weight="bold" />
                </a>
              </Flex>
              <p>siga-me nas redes sociais</p>
            </Flex>
          </S.SocialMediaContent>

          <Flex direction="column" justify="flex-end" gap="3.8rem">
            <motion.div variants={letterAnimation}>
              <Flex direction="column" align="stretch" gap="1rem">
                <Tag value="h1" />

                <S.HeroInfo>
                  <Flex direction="column">
                    <span>
                      <span className="highlight-word">Desenvolver</span> se
                    </span>
                    <span>tornou um hobby</span>
                    <span>favorito.</span>
                  </Flex>
                </S.HeroInfo>

                <Tag value="/h1" align="right" />
              </Flex>
            </motion.div>

            <motion.div variants={letterAnimation}>
              <Flex align="center" gap="6rem">
                <Flex gap="1rem" align="center">
                  <Image
                    src="/images/svg/logo.svg"
                    alt="Logotipo Daniel Silva • Sênior Front-End Developer"
                    title="Logotipo Daniel Silva • Sênior Front-End Developer"
                    aria-label="Logotipo dfsilva"
                    width={36}
                    height={36}
                    priority
                  />

                  <S.BottomTitle>
                    <Flex direction="column">
                      <span>Daniel Silva</span>
                      <span>Sênior Frontend Developer</span>
                    </Flex>
                  </S.BottomTitle>
                </Flex>

                <S.Technologies>
                  <Flex gap="6rem">
                    <span>React</span>
                    <span>Next</span>
                    <span>Node</span>
                  </Flex>
                </S.Technologies>
              </Flex>
            </motion.div>
          </Flex>
        </motion.div>
      </Container>
    </S.HeroContent>
  );
};

export default Hero;
