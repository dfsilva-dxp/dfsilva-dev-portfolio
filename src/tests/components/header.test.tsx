import { renderWithTheme } from '@/utils/tests/helpers';

import { Header } from '@/components';
import { screen } from '@testing-library/react';

describe('Header', () => {
  it('should render currectly header', () => {
    const { container } = renderWithTheme(<Header />);

    expect(
      screen.getByRole('img', { name: 'Logotipo dfsilva' })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
