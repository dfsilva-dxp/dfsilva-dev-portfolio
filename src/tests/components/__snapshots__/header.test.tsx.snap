// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render currectly header 1`] = `
.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  align-content: stretch;
  justify-content: space-between;
  flex-wrap: wrap;
  gap: 0;
}

.c0 {
  width: 100vw;
  padding: 2rem 6rem;
  background: transparent;
  position: fixed;
  top: 0;
  z-index: 10;
}

<header
  class="c0"
>
  <div
    class="c1"
    content="stretch"
    direction="row"
    display="flex"
    wrap="wrap"
  >
    <img
      alt="Logotipo Daniel Silva • Sênior Front-End Developer"
      aria-label="Logotipo dfsilva"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="36"
      src="/images/logo.svg"
      style="color: transparent;"
      title="Logotipo Daniel Silva • Sênior Front-End Developer"
      width="36"
    />
  </div>
</header>
`;
